lib/connections/svc.go

```diff
@@ -19,71 +19,89 @@ import (
 	"github.com/syncthing/syncthing/lib/config"
 	"github.com/syncthing/syncthing/lib/discover"
 	"github.com/syncthing/syncthing/lib/events"
-	"github.com/syncthing/syncthing/lib/model"
 	"github.com/syncthing/syncthing/lib/protocol"
-	"github.com/syncthing/syncthing/lib/relay"
-	"github.com/syncthing/syncthing/lib/relay/client"
 
 	"github.com/thejerf/suture"
 )
 
-type DialerFactory func(*url.URL, *tls.Config) (*tls.Conn, error)
-type ListenerFactory func(*url.URL, *tls.Config, chan<- model.IntermediateConnection)
+type intermediateConnection struct {
+	*tls.Conn
+	Type     string
+	Priority int
+}
+
+type Connection struct {
+	intermediateConnection
+	protocol.Connection
+}
+
+type genericDialer interface {
+	Dial(protocol.DeviceID, *url.URL, *tls.Config) (intermediateConnection, error)
+	Priority() int
+}
+
+type listenerFactory func(*url.URL, *tls.Config, chan intermediateConnection) genericListener
+
+type genericListener interface {
+	Serve()
+	Stop()
+	// Addresses which can potentially be mutated for peers to be able to connect.
+	// Also, a single listen address might yield multiple addresses due to finding
```

Comment:
So this is todo with relays,
for example we will listen on `dynamic+https://relays.syncthing.net`, but to connect to us you will have to use `relay://some.host.com:123` or potentially `[]string{relay://some.host.com:123, relay://some.otherhost.com:123`